cmd/tracee-ebpf/internal/printer/printer.go

```diff
@@ -198,16 +211,23 @@ func (p templateEventPrinter) Close() {
 }
 
 type jsonEventPrinter struct {
-	out io.WriteCloser
-	err io.WriteCloser
+	out      io.WriteCloser
+	err      io.WriteCloser
+	protocol bool
 }
 
 func (p jsonEventPrinter) Init() error { return nil }
 
 func (p jsonEventPrinter) Preamble() {}
 
 func (p jsonEventPrinter) Print(event trace.Event) {
-	eBytes, err := json.Marshal(event)
+	var err error
+	var eBytes []byte
+	if p.protocol {
+		eBytes, err = json.Marshal(event.ToProtocol())
+	} else {
+		eBytes, err = json.Marshal(event)
+	}
```

Comment:
I personally thought it would be useful for the case where we want to dump the only the payload itself, unrelated to sending to tracee-rules. Currently it seems though that the interface will change to receiving `protocol.Event` from the get go, so maybe we can just always print with the protocol.

Reply:
Resolved in latest commit